addons/wowl/doc/dropdown_component.md

```diff
@@ -0,0 +1,186 @@
+# Dropdown Component
+
+## Overview
+
+As dropdowns are common in Odoo, we decided to make a generic dropdown component.
+
+It contains all the logic of what you can usually expect a dropdown to behave.
+
+We made the choice to develop this simple component with as few style as possible, therefore it is your responsibility to make yours look awesome 😎!
+
+## Behind the scene
+
+A `<Dropdown/>` component is simply a `<div class="o_dropdown"/>` having a `<button/>` next to an unordered list (`<ul/>`). The button controls if the list is attached to the DOM or not.
+
+A `<DropdownItem/>` is simply a list item (`<li/>`) transporting a payload (which you'll receive back in a custom `dropdown-item-selected` event when the item get clicked).
+
+Illustration of what the final DOM could look like:
+
+```html
+<div class="o_dropdown">
+  <button>Click me to toggle the dropdown menu !</button>
+  <!-- following <ul/> list will or won't appear in the DOM depending on the state controlled by the button -->
+  <ul>
+    <li>Menu Item 1</li>
+    <li>Menu Item 2</li>
+  </ul>
+</div>
+```
+
+## Usage
+
+### Step 1: make it appear on your app
+
+In order to properly use a `<Dropdown/>` component, you need to populate two OWL slots:
+
+<dl>
+  <dt><strong>The <code>default</code> slot</strong></dt>
+  <dd>It contains the <strong>toggler elements of your dropdown</strong> and will take place inside your dropdown <code>&lt;button/></code> element.</dd>
+  <dt><strong>The <code>dropdown</code> slot</strong></dt>
+  <dd>
+    It contains the <strong>elements of the dropdown menu itself</strong> and will take place inside your dropdown <code>&lt;ul/></code> element.<br/>
+    Usually you will place at least one or more <code>&lt;DropdownItem/></code> elements in the <code>dropdown</code> slot.
+  </dd>
+</dl>
+
+So in your qweb template, you would write something like that:
+
+```xml
+<Dropdown>
+    <!-- "default" slot content should be defined here -->
+    Click me to toggle the dropdown menu !
+    <t t-set-slot="dropdown">
+      <!-- "dropdown" slot content should be defined here-->
+      <DropdownItem>Menu Item 1</DropdownItem>
+      <DropdownItem>Menu Item 2</DropdownItem>
+    </t>
+</Dropdown>
+```
+
+And in the DOM it would get translated similarly to:
+
+```xml
+<div class="o_dropdown">
+  <button>
+    <!-- "default" slot content will take place here -->
+    Click me to toggle the dropdown menu !
+  </button>
+
+  <ul>
+    <!-- "dropdown" slot content will take place here -->
+    <li>Menu Item 1</li>
+    <li>Menu Item 2</li>
+  </ul>
+</div>
+```
+
+### Step 2: make it react to clicks
+
+When a `<DropdownItem/>` get clicked, its parent `<Dropdown/>` will emit a custom `dropdown-item-selected` event containing the payload attached to the clicked item.
```

Comment:
gets clicked

Reply:
`dropdown-item-selected` or `dropdown-item-clicked`?

Reply:
- `dropdown-item-clicked` is fired by a DropdownItem
- its parent Dropdown catches it and may stop its propagation. In the other hand, its parent will fire a `dropdown-item-selected` event. This last one is the one I meant here in the doc.

See `onItemClicked` handler in dropdown.ts

Reply:
ok sorry

Reply:
but do we need two different events?

Reply:
Not really. It is fixed